### DISCUSSION TITLE
learn_one() and predict_one() do not seem to be working

### DISCUSSION THREAD

**Discussion by medha-chippa** [online-ml/river#593] [Q&A] [answered]
Im working on training a HoeffdingTreeClassifier using stream learning. Im using the learn_one() and predict_one() for the same. At the end of the training process, the list containing the predictions is empty. The data is in the form of a pandas dataframe(x data), and y is a pandas series. The iter_pandas() was used to convert the data to a form that is compatible with river. 

![image](https://user-images.githubusercontent.com/55135185/118689908-22147980-b825-11eb-9f15-66ea4cc3aa9b.png)
![image](https://user-images.githubusercontent.com/55135185/118690050-4b350a00-b825-11eb-8aa4-d87e74d3f15a.png)

Please help.

Thank you

**Comment by raphaelsty** [comment, marked as answer]
Hi @medha-chippa, is it possible to share some data to help you?  A small sample of x_bse_new and y?

**Reply by medha-chippa** [reply]
The same code when applied to the following data results in the same issue. I request you to please use the attached data. 
Thank you so much
[hepatitis_scikit_multiflow.zip](https://github.com/online-ml/river/files/6502943/hepatitis_scikit_multiflow.zip)

**Reply by medha-chippa** [reply]
x_bse_new = all the columns except for the last column (class), it is a pandas data frame
y = the last column(class), it is a pandas series

**Reply by raphaelsty** [reply]
Thank you, I reproduced your script and I didn't see any particular problem, maybe you have a problem when loading the data? I have an accuracy of 82.5% on the sample you shared. Maybe it will work if you copy paste my code?

```python
import pandas as pd
from river import stream
from river import tree
from river import metrics

X = pd.read_csv("hepatitis_scikit_multiflow.csv")
Y = X["Class"]
X = X.drop("Class", axis = 'columns')

model = tree.HoeffdingTreeClassifier()

X_y = stream.iter_pandas(X, Y)

metric = metrics.Accuracy()

y_preds = []

for x, y in X_y:
    
    y_pred = model.predict_one(x)
    
    model.learn_one(x, y)
    
    metric.update(y_pred=y_pred, y_true=y)
    
    y_preds.append(y_pred)

print(f"Accuracy: {metric.get()}")
```
Accuracy: 0.825

<img width="990" alt="image" src="https://user-images.githubusercontent.com/24591024/118692653-91c93b00-b80a-11eb-8661-0ab9566d0ede.png">

<img width="422" alt="image" src="https://user-images.githubusercontent.com/24591024/118692724-a4437480-b80a-11eb-9374-6518f8487250.png">

**Reply by medha-chippa** [reply]
Thank you so much @raphaelsty , i have been using google colab to work with my data. Do you think that must have caused the issue?

**Reply by raphaelsty** [reply]
Did it work? I don't think so, it should works well on collab :)

**Reply by medha-chippa** [reply]
It just worked, the problem was with using stream=stream.iter_pandas(x_train, y_train) and then using for x_i, y_i in "stream" to iterate over the stream. "stream" is a keyword, I had used the keyword to initialize my stream. 
Issue resolved. Thank you so much

**Reply by raphaelsty** [reply]
Cool! Happy that it work, have a good day @medha-chippa :)
